From: Josef Bacik <josef@redhat.com>
Date: Wed, 4 May 2011 11:11:17 -0400
Patch-mainline: yes
References: FATE#306586
Subject: [PATCH] Btrfs: set range_start to the right start in
 count_range_bits

In count_range_bits we are adjusting total_bytes based on the range we are
searching for, but we don't adjust the range start according to the range we are
searching for, which makes for weird results.  For example, if the range

[0-8192]

is set DELALLOC, but I search for 4096-8192, I will get back 4096 for the number
of bytes found, but the range_start will be 0, which makes it look like the
range is [0-4096].  So instead set range_start = max(cur_start, state->start).
This makes everything come out right.  Thanks,

Signed-off-by: Josef Bacik <josef@redhat.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/extent_io.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.32-SLE11-SP2/fs/btrfs/extent_io.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/fs/btrfs/extent_io.c
+++ linux-2.6.32-SLE11-SP2/fs/btrfs/extent_io.c
@@ -1451,7 +1451,7 @@ u64 count_range_bits(struct extent_io_tr
 			if (total_bytes >= max_bytes)
 				break;
 			if (!found) {
-				*start = state->start;
+				*start = max(cur_start, state->start);
 				found = 1;
 			}
 			last = state->end;
